Core/Src/pedals.c

```diff
@@ -0,0 +1,474 @@
+/**
+ * @file processing.c
+ * @author Scott Abramson
+ * @brief Definitions for tasks for processing data.
+ * @version 0.1
+ * @date 2024-08-04
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#include "pedals.h"
+#include "state_machine.h"
+#include "queues.h"
+#include "cerb_utils.h"
+#include "nero.h"
+#include "can_handler.h"
+#include "cerberus_conf.h"
+#include "dti.h"
+#include "queues.h"
+#include "serial_monitor.h"
+#include "ams.h"
+#include "emrax.h"
+#include "monitor.h"
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+#include <math.h>
+#include <stdlib.h>
+
+/* DO NOT ATTEMPT TO SEND TORQUE COMMANDS LOWER THAN THIS VALUE */
+#define MIN_COMMAND_FREQ  60 /* Hz */
+#define MAX_COMMAND_DELAY 1000 / MIN_COMMAND_FREQ /* ms */
+
+static float torque_limit_percentage = 1.0;
+
+/* Parameters for the pedal monitoring task */
+#define MAX_ADC_VAL_12b	  4096
+#define PEDAL_DIFF_THRESH 30
+#define PEDAL_FAULT_TIME  500 /* ms */
+
+static bool brake_state = false;
+osMutexId_t brake_mutex;
+
+enum { ACCELPIN_2, ACCELPIN_1, BRAKEPIN_1, BRAKEPIN_2 };
+
+void increase_torque_limit()
+{
+	if (torque_limit_percentage + 0.1 > 1) {
+		torque_limit_percentage = 1;
+	} else {
+		torque_limit_percentage += 0.1;
+	}
+}
+
+void decrease_torque_limit()
+{
+	if (torque_limit_percentage - 1 < 0) {
+		torque_limit_percentage = 0;
+	} else {
+		torque_limit_percentage -= 0.1;
+	}
+}
+
+void set_brake_state(bool new_brake_state)
+{
+	osMutexAcquire(brake_mutex, osWaitForever);
```

Comment:
make this MUTEX_TIMEOUT.  Also do this throughout.

Reply:
MUTEX_TIMEOUT in pdu.c is equal to osWaitForever.